File: rest-common/src/main/java/org/chronopolis/rest/models/repair/AuditStatus.java
package org.chronopolis.rest.models.repair;

import com.google.common.collect.ImmutableListMultimap;

/**
 * Encapsulate some states for Repair audits
 *
 * Created by shake on 2/23/17.
 */
public enum AuditStatus {
    PRE, AUDITING, SUCCESS, FAIL;

    public static ImmutableListMultimap<String, AuditStatus> statusByGroup() {
        return new ImmutableListMultimap.Builder<String, AuditStatus>()
                .put("Pending", AuditStatus.PRE)
                .put("Active", AuditStatus.AUDITING)
                .put("Success", AuditStatus.SUCCESS)
                .put("Failure", AuditStatus.FAIL)
                .build();
    }
}

File: rest-models/src/main/kotlin/org/chronopolis/rest/kot/models/enums/AuditStatus.kt
package org.chronopolis.rest.kot.models.enums

import com.google.common.collect.ImmutableListMultimap

enum class AuditStatus {
    PRE, AUDITING, SUCCESS, FAIL;

    companion object {
        fun statusByGroup(): ImmutableListMultimap<String, AuditStatus> =
                ImmutableListMultimap.Builder<String, AuditStatus>()
                        .put("Pending", PRE)
                        .put("Active", AUDITING)
                        .put("Success", SUCCESS)
                        .put("Failure", FAIL)
                        .build()
    }
}